apps/_components/src/CodeBlock/CodeBlock.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import { Button } from '@digdir/designsystemet-react';
+import cl from 'clsx/lite';
+import { useState } from 'react';
+import { CodeBlock as ReactCodeBlock } from 'react-code-block';
+import classes from './CodeBlock.module.css';
+
+export type CodeBlockProps = {
+  children: string;
+  className?: string;
+  /**
+   * Supported languages:
+   *  "markup"
+   *  "jsx"
+   *  "tsx"
+   *  "swift"
+   *  "kotlin"
+   *  "objectivec"
+   *  "js-extras"
+   *  "reason"
+   *  "rust"
+   *  "graphql"
+   *  "yaml"
+   *  "go"
+   *  "cpp"
+   *  "markdown"
+   *  "python"
+   *  "json"
+   */
+  language?: string;
```

Comment:
```suggestion
/** 
 * Supported languages
 */
  language?:
    | 'markup'
    | 'jsx'
    | 'tsx'
    | 'swift'
    | 'kotlin'
    | 'objectivec'
    | 'js-extras'
    | 'reason'
    | 'rust'
    | 'graphql'
    | 'yaml'
    | 'go'
    | 'cpp'
    | 'markdown'
    | 'python'
    | 'json'
    | 'text';
```

Reply:
We need to allow string here as well.

Reply:
Why and what kind of strings?

Reply:
Why not just add the "other" strings?

Reply:
huh? Due to the class that comes from mdx, we have to transform it into what language it is.
Typescript sees this as just a string
```tsx
pre: ({
      children: {
        props: { children = '', className = '' },
      },
    }) => {
      return (
        <CodeBlock language={className.replace('language-', '')}>
          {children}
        </CodeBlock>
      );
    }
```

`"shell"`, used in the token modal does not give a type error, but it defaults to plain text if the language is not supported.

Reply:
aha, now I understand, so its for the integration with mdx and code ticks to work. Maybe a comment on the type in CodeBlock or an issue to explain this for future selfs? :)

Reply:
sure I'll add a comment about it